devboard/maixbit/examples/pwm/main.go

```diff
@@ -36,36 +36,35 @@ func main() {
 	ch2.Setup(fpioa.TIMER0_TOGGLE1 | fpioa.EnOE | fpioa.DriveH34L23)
 	ch3.Setup(fpioa.TIMER0_TOGGLE3 | fpioa.EnOE | fpioa.DriveH34L23)
 
+	// Peripheral
+	p := timer.TIMER(0)
+
 	// Driver instance
-	r := timer.NewDriver(timer.TIMER(0), 0)
-	g := timer.NewDriver(timer.TIMER(0), 1)
-	b := timer.NewDriver(timer.TIMER(0), 2)
+	r := timer.PWM(p.Channel(0))
+	g := timer.PWM(p.Channel(1))
+	b := timer.PWM(p.Channel(2))
 
 	// Set frequency now so we don't have to wait around for the first period to expire
 	r.SetFrequency(freq, .5)
 	g.SetFrequency(freq, .5)
 	b.SetFrequency(freq, .5)
 
-	// Animate duty cycle
-	go func() {
-		dc := 0.0
-		for {
-			r.SetFrequency(freq, math.Abs(math.Sin(dc)))
-			g.SetFrequency(freq, math.Abs(math.Sin(dc+math.Pi)))
-			b.SetFrequency(freq, math.Abs(math.Cos(dc)))
-			dc += 0.02
-			time.Sleep(10 * time.Millisecond)
-		}
-	}()
-
 	// Enable the timer and PWM function
-	r.EnablePWM()
-	g.EnablePWM()
-	b.EnablePWM()
+	r.Enable()
+	g.Enable()
+	b.Enable()
 
 	// Enable an ISR for the green channel
-	g.EnableISR()
+	g.EnableIRQ()
 	irq.TIMER0A.Enable(rtos.IntPrioLow, irq.M0)
```

Comment:
I think you should comment that this is only for IRQ demonstration and don't need to use PWM. Or make this example simple and avoid IRQs entirely in it. You can provide the separate pwmirq or timerirq example.

I still don't understand the relation between PWM and IRQs here and why there are A and B interrupts for timers. Can you shed some light on it?